definitions/npm/styled-components_v2.x.x/flow_v0.53.x-/styled-components_v2.x.x.js

```diff
@@ -0,0 +1,332 @@
+// @flow
+
+type $npm$styledComponents$Interpolation = ((executionContext: Object) => string) | string | number;
+type $npm$styledComponents$NameGenerator = (hash: number) => string;
+
+type $npm$styledComponents$TaggedTemplateLiteral<R> = {| (Array<string>, $npm$styledComponents$Interpolation): R |};
+
+type $npm$styledComponents$ReactComponentFunctional<Props: {}> = Props => React$Element<*>;
+type $npm$styledComponents$ReactComponentClass<Props, DefaultProps = *> = Class<React$Component<DefaultProps & Props, *>>
+
+type $npm$styledComponents$ReactComponentUnion<Props> =
```

Comment:
I tried replacing it but ran across several issues that keep compounding. I'd love for someone to take a look at it.

My best guess is that there's a buggy implementation somewhere in there or test mismatch but I've yet to discover it.